icu4c/source/common/charstr.cpp

```diff
@@ -141,6 +141,38 @@ CharString &CharString::append(const char *s, int32_t sLength, UErrorCode &error
     return *this;
 }
 
+CharString &CharString::appendNumber(int32_t number, UErrorCode &status) {
```

Comment:
It would be good to ask Markus to review this addition to CharString, if he hasn't already. He has opinions on how it should evolve.